tests/amqpconnection_ssl_by_cert.phpt

```diff
@@ -0,0 +1,156 @@
+$cnn->setHost('localhost');
+$cnn->setPort(5671);
+$cnn->setCACert("/home/vagrant/php-amqp/provision/test_certs/testca/cacert.pem");
+$cnn->setCert("/home/vagrant/php-amqp/provision/test_certs/sasl-client/cert.pem");
+$cnn->setKey("/home/vagrant/php-amqp/provision/test_certs/sasl-client/key.pem");
+$cnn->setSaslMethod(1);
+$cnn->setVerify(true);
+var_dump($cnn);
+
+$cnn->connect();
+
+echo ($cnn->isConnected() ? 'connected' : 'disconnected'), PHP_EOL;
+
+//sleep(100);
+
+?>
+--EXPECT--
+object(AMQPConnection)#1 (16) {
+  ["login":"AMQPConnection":private]=>
+  string(5) "guest"
+  ["password":"AMQPConnection":private]=>
+  string(5) "guest"
+  ["host":"AMQPConnection":private]=>
+  string(20) "rabbitmq.example.org"
+  ["vhost":"AMQPConnection":private]=>
+  string(1) "/"
+  ["port":"AMQPConnection":private]=>
+  int(5671)
+  ["read_timeout":"AMQPConnection":private]=>
+  float(0)
+  ["write_timeout":"AMQPConnection":private]=>
+  float(0)
+  ["connect_timeout":"AMQPConnection":private]=>
+  float(0)
+  ["channel_max":"AMQPConnection":private]=>
+  int(256)
+  ["frame_max":"AMQPConnection":private]=>
+  int(131072)
+  ["heartbeat":"AMQPConnection":private]=>
+  int(0)
+  ["cacert":"AMQPConnection":private]=>
+  string(61) "/home/vagrant/php-amqp/provision/test_certs/testca/cacert.pem"
+  ["key":"AMQPConnection":private]=>
+  string(63) "/home/vagrant/php-amqp/provision/test_certs/sasl-client/key.pem"
+  ["cert":"AMQPConnection":private]=>
+  string(64) "/home/vagrant/php-amqp/provision/test_certs/sasl-client/cert.pem"
+  ["verify":"AMQPConnection":private]=>
+  bool(true)
+  ["sasl_method":"AMQPConnection":private]=>
+  int(1)
+}
+connected
+
+object(AMQPConnection)#2 (16) {
+  ["login":"AMQPConnection":private]=>
+  string(5) "guest"
+  ["password":"AMQPConnection":private]=>
+  string(5) "guest"
+  ["host":"AMQPConnection":private]=>
+  string(9) "localhost"
+  ["vhost":"AMQPConnection":private]=>
+  string(1) "/"
+  ["port":"AMQPConnection":private]=>
+  int(5671)
+  ["read_timeout":"AMQPConnection":private]=>
+  float(0)
+  ["write_timeout":"AMQPConnection":private]=>
+  float(0)
+  ["connect_timeout":"AMQPConnection":private]=>
+  float(0)
+  ["channel_max":"AMQPConnection":private]=>
+  int(256)
+  ["frame_max":"AMQPConnection":private]=>
+  int(131072)
+  ["heartbeat":"AMQPConnection":private]=>
+  int(0)
+  ["cacert":"AMQPConnection":private]=>
+  string(61) "/home/vagrant/php-amqp/provision/test_certs/testca/cacert.pem"
+  ["key":"AMQPConnection":private]=>
+  string(63) "/home/vagrant/php-amqp/provision/test_certs/sasl-client/key.pem"
+  ["cert":"AMQPConnection":private]=>
+  string(64) "/home/vagrant/php-amqp/provision/test_certs/sasl-client/cert.pem"
+  ["verify":"AMQPConnection":private]=>
+  bool(true)
+  ["sasl_method":"AMQPConnection":private]=>
+  int(1)
+}
+
+Fatal error: Uncaught AMQPConnectionException: Socket error: could not connect to host. in /home/vagrant/php-amqp/tests/amqpconnection_ssl_by_cert.php:68
```

Comment:
Is it expected here?

Reply:
@pinepain 
yes, it is, because we are connecting to localhost with host verification turned on
( verify = true )